### ISSUE TITLE
MD029/ol-prefix: Continuous list with more than 10 elements

### ISSUE THREAD

**Issue by Mazorius** [DavidAnson/markdownlint#1237]
Hi,

I created a list with 14 entries.
All of them have a code example in it.

See example:

```markdown
1. ...
2. ...
   1. ...
   2. ...
3. ...:

   ```shell
   code example
   ```

4. ...:

   ```shell
   code example
   ```

5. ...:

   ```shell
   code example
   ```

6. ...:

   ```shell
   code example
   ```

7. ...:

   ```shell
   code example
   ```

8. ...
9. ...:

   ```shell
   code example
   ```

10. ...:

   ```shell
   code example
   ```

11. ...:

    ```shell
    code example
    ```

12. ...:

    ```shell
    code example
    ```

13. ...:

    ```shell
    code example
    ```

14. ...:

    ```shell
    code example
    ```

```

I am using markdownlint via pre-commit:

```yaml
  - repo: https://github.com/igorshubovych/markdownlint-cli
    rev: v0.41.0
    hooks:
      - id: markdownlint
```

IntelliJIdea successfully recognises the list item and names them correctly as you see above.

From markdownlint I got this error:

```log
*.md:N:1 MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 11; Style: 1/2/3]
*.md:N:1 MD029/ol-prefix Ordered list item prefix [Expected: 2; Actual: 12; Style: 1/2/3]
*.md:N:1 MD029/ol-prefix Ordered list item prefix [Expected: 3; Actual: 13; Style: 1/2/3]
*.md:N:1 MD029/ol-prefix Ordered list item prefix [Expected: 4; Actual: 14; Style: 1/2/3]
```

Is there an issue with higher numbers?

**Comment by Mazorius** [comment]
Was a tap vs space issue in point ten. everything works fine

**Comment by zachliu** [comment]
i think you've discovered something here :thinking: 

i had to use double tab (4 spaces) for all items >= 10 otherwise i got the same error on 11, 12, ...etc.
`MD029/ol-prefix Ordered list item prefix [Expected: 1; Actual: 11; Style: 1/2/3]`

for markdown file type, there are no "hard" tabs in my case. i have `shiftwidth=2 softtabstop=2` but i had to use 3 spaces for code blocks indentation otherwise they don't work 

``````markdown
# Test

1. ...
2. ...
   1. three spaces here
   2. three spaces here
3. ...:

   ```bash
   three spaces here
   ```

4. ...:

   ```bash
   three spaces here
   ```

5. ...:

   ```bash
   three spaces here
   ```

6. ...:

   ```bash
   three spaces here
   ```

7. ...:

   ```bash
   three spaces here
   ```

8. ...:

   ```bash
   three spaces here
   ```

9. ...:

   ```bash
   three spaces here
   ```

10. ...:

    ```bash
    double tab (four spaces) here
    ```

11. ...:

    ```bash
    double tab (four spaces) here
    ```

12. ...:

    ```bash
    double tab (four spaces) here
    ```
``````

**Comment by zachliu** [comment]
@DavidAnson good afternoon sir. is this related to https://github.com/DavidAnson/markdownlint/issues/38 and https://github.com/DavidAnson/markdownlint/issues/125? so i guess when the index number >= 10, we have to use 4 spaces on code fences instead of 3. am i correct?

**Comment by DavidAnson** [comment]
The person who opened this issue closed it themselves saying everything was fine. If you are seeing something that seems wrong, please provide a simple example and explain what happens versus what you expect to happen.

**Comment by zachliu** [comment]
oh ok, i thought my [comments](https://github.com/DavidAnson/markdownlint/issues/1237#issuecomment-2327289903) up there (with an example) was pretty much self-explanatory
